packages/bpk-component-button/src/BpkButtonDestructive.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import React from 'react';
+
+import { type Props, defaultProps, propTypes } from './common-types';
+import BpkButtonBase from './BpkButtonBase';
+import STYLES from './BpkButtonDestructive.scss';
+
+// This was originally depended upon from the bpk-react-utils package, however
+// we decided to inline it in this particular component so as not to bloat the
+// the bundles of consumers who are not yet on webpack 2
+// We'll revisit this again soon.
+const cssModules = (styles = {}) => className =>
+  styles[className] ? styles[className] : className;
+
+const getClassName = cssModules(STYLES);
+
+const BpkButtonDestructive = (props: Props) => {
+  const { className, ...rest } = props;
+
+  const classNames = [getClassName('bpk-button--destructive')];
+
+  if (className) {
+    classNames.push(className);
+  }
+  const classNamesFinal = classNames.join(' ');
+
+  return <BpkButtonBase className={classNamesFinal} {...rest} />;
+};
+
+BpkButtonDestructive.propTypes = propTypes;
```

Comment:
can we make this ```{ ...propTypes }```. This can be a bit dangerous since all buttons are referencing the same object and changing it will affect all buttons.

We should also remove the `type` prop from the specific buttons as they don't need it